practice-app/server/src/routes/chess/chess.controller.js

```diff
@@ -23,25 +27,145 @@ const ChessController = {
             },
         };
 
-        const response = (await axios.post(url, payload, headers)).data;
+        try {
+            const response = (await axios.post(url, payload, headers)).data;
 
-        if (response && response.id) {
-            const game = await ChessModel.createGame(
-                response.id,
-                selectedColor
-            );
-            if (game) {
-                return res.status(200).json({
+            if (response && response.id) {
+                const game = await ChessGame.create({
                     game_id: response.id,
                     player_color: selectedColor,
                 });
+                if (game) {
+                    return res.status(200).json({
+                        game_id: response.id,
+                        player_color: selectedColor,
+                    });
+                }
             }
+        } catch (e) {
+            console.log(e);
         }
 
         return res.status(500).json({
             message: "Could not initiate a game against the AI",
         });
     },
+    makeMove: async function (req, res) {
+        // Get user input
+        const { gameId, moveStr } = req.body;
+
+        const url = `https://lichess.org/api/board/game/${gameId}/move/${moveStr}`;
+
+        const payload = {};
+        const headers = {
+            headers: {
+                Authorization: `Bearer ${process.env.LICHESS_TOKEN}`,
+            },
+        };
+
+        try {
+            const response = (await axios.post(url, payload, headers)).data;
+
+            if (response && response.ok) {
+                return res.status(200).json({
+                    ok: true,
+                });
+            }
+        } catch (e) {
+            return res
```

Comment:
Your way of handling errors is not consistent between `makeMove` and `createGame` functions. In `createGame` function, when you catch an error you just log it and return 500 with a generic error message. However, in `makeMove` function you return 400 in the catch block with a generic message and return 500 with the response from the Lichess api, which has credentials in the header section. I think this function will return either 200 or 400, due to try catch block but I wanted to be sure about this.

Reply:
returning 400 in catch block with a message. removed 500.